docarray/index/backends/elastic.py

```diff
@@ -0,0 +1,546 @@
+import os
+import uuid
+import warnings
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import (
+    Any,
+    Dict,
+    Generator,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import numpy as np
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import parallel_bulk
+from pydantic import parse_obj_as
+
+import docarray.typing
+from docarray import BaseDocument
+from docarray.index.abstract import (
+    BaseDocumentIndex,
+    _ColumnInfo,
+    _FindResultBatched,
+    _raise_not_composable,
+)
+from docarray.typing import AnyTensor
+from docarray.typing.tensor.ndarray import NdArray
+from docarray.utils.find import _FindResult
+from docarray.utils.misc import is_tf_available, is_torch_available
+
+TSchema = TypeVar('TSchema', bound=BaseDocument)
+T = TypeVar('T', bound='ElasticDocIndex')
+
+ELASTIC_PY_VEC_TYPES: List[Any] = [list, tuple, np.ndarray]
+
+if is_torch_available():
+    import torch
+
+    ELASTIC_PY_VEC_TYPES.append(torch.Tensor)
+
+if is_tf_available():
+    import tensorflow as tf  # type: ignore
+
+    from docarray.typing import TensorFlowTensor
+
+    ELASTIC_PY_VEC_TYPES.append(tf.Tensor)
+    ELASTIC_PY_VEC_TYPES.append(TensorFlowTensor)
+
+
+class ElasticDocIndex(BaseDocumentIndex, Generic[TSchema]):
```

Comment:
I think we should call this one the `ElasticV7DocIndex` and the other one simply `ElasticDocIndex`, since v7 is a legacy version.